### PR TITLE
Update OIDC documentation adding a warning for Okta configuration

### PR DESCRIPTION
**Pull request by pipo02mix**
### Please remember to

- [x] Give the PR a meaningful title -- it will be shown in the release notes
- [ ] Add (or remove) [user questions](https://github.com/giantswarm/docs/blob/main/CONTRIBUTING.md#front-matter) associated with any updated docs

### CODE REVIEW

File: src/content/advanced/configure-dex-in-your-cluster/index.md [hunk 1]
```diff
@@ -92,13 +92,6 @@ metadata:
 data:
   values: |
     isWorkloadCluster: true
-    services:
-      kubernetes:
-        api:
-          caPem: |
-            -----BEGIN CERTIFICATE-----
-            M...=
-            -----END CERTIFICATE-----
```
Comment (pipo02mix) [inline src/content/advanced/configure-dex-in-your-cluster/index.md:101]: this is now automatically injected in last versions

File: src/content/advanced/configure-dex-in-your-cluster/index.md [hunk 2]
```diff
@@ -193,9 +176,46 @@ data:
...
+            clientID: <CLIENT-ID-SET-IN-YOUR-IdP>
+            clientSecret: <CLIENT-SECRET-SET-IN--YOUR-IdP>
+            insecureEnableGroups: true
+            getUserInfo: true
+            scopes:
+            - email
+            - groups
+            - profile
+            issuer: https://<OKTA_OIDC_ENDPOINT>
+            redirectURI: https://dex.<CLUSTERID>.<BASEDOMAIN>/callback
```
Comment (kopiczko) [inline src/content/advanced/configure-dex-in-your-cluster/index.md:211]: `insecureSkipEmailVerified: true` is missng

File: src/content/advanced/configure-dex-in-your-cluster/index.md (hunk 1 above)
Comment (marians) [reply src/content/advanced/configure-dex-in-your-cluster/index.md:101]: ❤️

File: src/content/advanced/configure-dex-in-your-cluster/index.md [hunk 3]
```diff
@@ -149,12 +135,16 @@ data:
...
           connectorConfig: >-
             clientID: <CLIENT-ID-SET-IN-YOUR-IdP>
             clientSecret: <CLIENT-SECRET-SET-IN--YOUR-IdP>
-            tenant: <TENANT-SET-SET-IN--YOUR-IdP>
+            loadAllGroups: false
             redirectURI: https://dex.<CLUSTERID>.<BASEDOMAIN>/callback
+            orgs:
+            - name: main-customer-org
+              teams:
+              - team-infra
```
Comment (marians) [inline src/content/advanced/configure-dex-in-your-cluster/index.md:147]: Why this change?

Comment (marians) [reply src/content/advanced/configure-dex-in-your-cluster/index.md:147]: Oh, I see. It's supposed to be the GitHub example, but used the wrong `connectorType`.

The rest with `loadAllGroups: false` however I'd still like to understand. Why not load all group memberships?

File: src/content/advanced/configure-dex-in-your-cluster/index.md [hunk 4]
```diff
@@ -193,9 +176,46 @@ data:
...
+            - groups
+            - profile
+            issuer: https://<OKTA_OIDC_ENDPOINT>
+            redirectURI: https://dex.<CLUSTERID>.<BASEDOMAIN>/callback
+```
+
 {{< /tab >}}
 {{< /tabs >}}
 
+__Warning__: With `oidc` connector you might need to add `getUserInfo` in the connector configuration to force a second called to IdP in order to get groups (for example required by Okta). More info [here](https://github.com/dexidp/dex/issues/1065).
```
Comment (marians) [inline src/content/advanced/configure-dex-in-your-cluster/index.md:217]:
```suggestion
__Warning__: With `oidc` connector you might need to add `getUserInfo` in the connector configuration to force a second call to the identity provider in order to get groups. This is required for example by Okta. More info on this can be found in [dexipd/dex#1065](https://github.com/dexidp/dex/issues/1065).
```

File: src/content/advanced/configure-dex-in-your-cluster/index.md [hunk 5]
```diff
@@ -193,9 +176,46 @@ data:
...
+      customer:
+        enabled: true
+        connectors:
+        - id: customer
+          connectorName: test
+          connectorType: oidc
+          connectorConfig: >-
+            clientID: <CLIENT-ID-SET-IN-YOUR-IdP>
+            clientSecret: <CLIENT-SECRET-SET-IN--YOUR-IdP>
+            insecureEnableGroups: true
```
Comment (pipo02mix) [inline src/content/advanced/configure-dex-in-your-cluster/index.md:204]:
```suggestion
            insecureEnableGroups: true
            insecureSkipEmailVerified: true
```

File: src/content/advanced/configure-dex-in-your-cluster/index.md (hunk 3 above)
Comment (pipo02mix) [reply src/content/advanced/configure-dex-in-your-cluster/index.md:147]: yeah it was a mistake and I have just realized. 

IMO it is a good default, so define which orgs you want to give access instead of pulling all

File: src/content/advanced/configure-dex-in-your-cluster/index.md [hunk 6]
```diff
@@ -193,9 +176,47 @@ data:
...
+            insecureSkipEmailVerified: true
```
Comment (kopiczko) [inline src/content/advanced/configure-dex-in-your-cluster/index.md:205]: Sorry @pipo02mix I didn't realise this is a generic guide. In that case `insecureSkipEmailVerified` should go to the warning section. I think same applies to `insecureEnableGroups`. I'm not even sure if we needed to add this in our last case.

Comment (pipo02mix) [inline src/content/advanced/configure-dex-in-your-cluster/index.md:205]:
```suggestion
```

File: src/content/advanced/configure-dex-in-your-cluster/index.md (hunk 3 above)
Comment (marians) [reply src/content/advanced/configure-dex-in-your-cluster/index.md:147]: OK for me

**Review by marians — APPROVED** [review]

**Review by kopiczko — APPROVED** [review]

**Comment by marians** [comment]
@pipo02mix Wanna merge?

**Review by MarcelMue — APPROVED** [review]
